R/variables.R

```diff
@@ -321,6 +321,8 @@ initial_immunity <- function(
   ) {
   if (!is.null(eq)) {
     age <- age / 365
+    # older ages must be assigned within the equilibrium age bounds
+    age[age>=max(eq[[1]][, 'age'])] <- max(eq[[1]][, 'age'])-0.01
```

Comment:
Would adding a `max_age` parameter to `calculate_initial_ages()` and overwriting any initial ages > max_age in that function be a cleaner and clearer application?

Reply:
That sounds reasonable - Michael's model uses a truncated exponential distribution to draw ages anyway, so it doesn't have the same issue. It probably wouldn't be unreasonable to switch over to that. @giovannic ?

Reply:
hmm, it seems right to sample from a truncated exponential here too. From a quick look at mrc-ide/malariasimulation, it would be consistent.

You could use the maximum age in EQUILIBRIUM_AGES as your upper bound. `rtexp` in utils.R could do the sampling for you fyi.

Reply:
Sure thing. So if we sample ages from the truncated exponential, there won't be an issue with individuals being outside the eq solution.

My only concern is the way that the initial age distribution would be different to the age distribution at model equilibrium, although maybe this wouldn't be a problem. MW also limits ages to the max equilibrium solution age, so maintains a truncated exp dist throughout, but this might not be consistent with our model...?

Reply:
It should have very minimal impact as we're changing a very small proportion of the population